Neos.Neos/Classes/AssetUsage/Service/GlobalAssetUsageService.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Neos\Neos\AssetUsage\Service;
+
+use Neos\ContentRepository\Core\ContentRepository;
+use Neos\ContentRepository\Core\Factory\ContentRepositoryId;
+use Neos\ContentRepository\Core\Factory\ContentRepositoryServiceInterface;
+use Neos\ContentRepositoryRegistry\ContentRepositoryRegistry;
+use Neos\Neos\AssetUsage\AssetUsageFinder;
+use Neos\Neos\AssetUsage\Dto\AssetUsageFilter;
+use Neos\Neos\AssetUsage\Dto\AssetUsages;
+use Neos\Flow\Annotations as Flow;
+use Neos\Neos\AssetUsage\Projection\AssetUsageRepository;
+
+/**
+ * @internal
+ */
+class GlobalAssetUsageService implements ContentRepositoryServiceInterface
+{
+    /**
+     * @var array<string, mixed>
+     */
+    #[Flow\InjectConfiguration(path: "AssetUsage")]
+    protected array $flowSettings;
+
+    /**
+     * @var array<string, ContentRepository>
+     */
+    protected ?array $repositories = null;
+
+    public function __construct(
+        protected readonly ContentRepositoryRegistry $contentRepositoryRegistry,
+    ) {
+    }
+
+    public function findAssetUsageByAssetId(string $assetId): AssetUsages
+    {
+        $assetUsages = $this->withAllRepositories(
+            function (ContentRepository $repository) use ($assetId) {
+                $filter = AssetUsageFilter::create()
+                    ->withAsset($assetId)
+                    ->includeVariantsOfAsset()
+                    ->groupByNode();
+
+                return $repository->projectionState(AssetUsageFinder::class)->findByFilter($filter);
+            }
+        );
+
+        return AssetUsages::fromArrayOfAssetUsages($assetUsages);
+    }
+
+    public function removeAssetUsageByAssetId(string $assetId): void
+    {
+        $this->withAllRepositories(
+            fn(AssetUsageRepository $repository) => $repository->removeAsset($assetId)
```

Comment:
`$repository` is an instance of `ContentRepository` according to the code and there is no `ContentRepository::removeAsset()` or what did I miss?

Reply:
You are right, this is a bug. I fixed it. Just wondering, why this error didn't apper during testing.